superdesk/io/subjectcodes.py

```diff
@@ -45,22 +43,24 @@ def get_items(self):
         """
         items = []
         for code in sorted(self.subjects):
-            items.append({'qcode': code, 'name': self.subjects[code], 'parent': self._get_parent_code(code)})
+            items.append({'qcode': code, 'name': self.subjects[code], 'parent': get_parent_subjectcode(code)})
         return items
 
-    def _get_parent_code(self, code):
-        """Compute parent code for iptc newscode.
 
-        :param code: iptc newscode without `subj:` prefix
-        """
-        parent_code = None
-        if not self.newscode_pattern.match(code):
-            return parent_code
-        if code[-3:] != '000':
-            parent_code = code[:5] + '000'
-        elif code[2:5] != '000':
-            parent_code = code[:2] + '000000'
+def get_parent_subjectcode(code):
+    """Compute parent code for iptc newscode.
+
+    :param code: iptc newscode without `subj:` prefix
+    """
+    newscode_pattern = re.compile('[0-9]{8,9}')
```

Comment:
I would put it somewhere on top of the file like `NEWSCODE_PATTERN = re.compile(...)` so we don't compile it all the time